File: C++/armstrong.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int num;
    cin>>num;

    int count = 0, numcopy = num, sum = 0, rem;
    while(numcopy != 0){
        numcopy = numcopy /10;
        count++;
    }
    numcopy = num;
    while(numcopy != 0){
        rem = numcopy % 10;
        sum = sum + pow(rem, count);
        numcopy = numcopy / 10;
    }
    if(sum == num){
        cout<<"Number is Armstrong.";
    }
    else{
        cout<<"Number is not Armstrong.";
    }
    return 0;
}

File: Python/armstrong.py
# An Armstrong number is a number the sum of whose digits raised to the power of total number of digits is the number itself.

# Function to check if given number is Armstrong or not.
def armstrong(val):
    if int(val) >= 0:
        sum = 0
        for i in val:
            sum += int(i)**len(val)
        if sum == int(val):
            return 1
    return 0


num = input("Enter an integer: ")

if armstrong(num):
    print(num + " is an Armstrong number.")
else:
    print(num + " is not an Armstrong number")
